include/aca_security_group.h

```diff
@@ -0,0 +1,188 @@
+//
+// Created by Administrator on 2020/10/12.
+//
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ALCOR_CONTROL_AGENT_SECURITY_GROUP_H
+#define ALCOR_CONTROL_AGENT_SECURITY_GROUP_H
+#include <stdint.h>
+#include <string>
+#include <map>  
+#include <vector>  
+#include <set>  
+
+using namespace std;
+
+namespace aca_security_group
+{
+
+enum OperationType {
+	CREATE=1, 
+	UPDATE, 
+	DELETE,
+	UNKNOWN_OPERATION,
+};
+
+enum Direction {
+	INGRESS=1, 
+	EGRESS, 
+	UNKNOWN_DIRECTION,
+};
+
+enum Ethertype {
+	IPV4=0x0800, 
+	ARP=0x0806, 
+	IPV6=0x86dd,
+	UNKNOWN_ETHERTYPE,
+};
+
+enum Protocol {
+	TCP=6, 
+	UDP=17, 
+	ICMP=1,
+	UNKNOWN_PROTO
+};
+
+class Aca_Security_Group;
+
+class Aca_Security_Group_Rule {
+public:
+    static Aca_Security_Group_Rule &get_instance();
```

Comment:
ACA uses .clang-format to do automatic formatting. I have the setting in my IDE to auto-format on file save. Please see if you can do similar settings in your IDE.